library/std/src/sys/wasm/mutex_atomics.rs

```diff
@@ -26,11 +26,14 @@ impl Mutex {
 
     pub unsafe fn lock(&self) {
         while !self.try_lock() {
-            let val = wasm32::memory_atomic_wait32(
-                self.ptr(),
-                1,  // we expect our mutex is locked
-                -1, // wait infinitely
-            );
+            // SAFETY: the caller must uphold the safety contract for `i32_atomic_wait`.
+            let val = unsafe {
+                wasm32::i32_atomic_wait(
```

Comment:
I think this is a rebase failure -- this function seems to have been renamed to memory_atomic_wait32 on master. (I think there's one or two more cases where you have that).